MonkeyBot/Modules/Polls/PollService.cs

```diff
@@ -28,50 +31,77 @@ public async Task AddAndStartPollAsync(Poll poll)
+                {
+                    var user = btnClick.Result.User;
+                    var answerId = btnClick.Result.Id;
+
+                    var answerDb = poll.PossibleAnswers.First(x => x.Id == answerId);
+                    
+                    answerDb.UpdateCount(user.Id);
+                    _dbContext.Update(poll);
+                    await  _dbContext.SaveChangesAsync(cancelTokenSource.Token);
+                    
+                    embedBuilder = embedBuilder
+                        .WithDescription(PollMessageUpdater.BuildDescription(poll.PossibleAnswers));
+                    msgBuilder = msgBuilder
+                        .WithEmbed(embedBuilder.Build());
+                    var interactionBuilder = new DiscordInteractionResponseBuilder(msgBuilder);
+                    await btnClick.Result.Interaction.CreateResponseAsync(InteractionResponseType.UpdateMessage, interactionBuilder);
+                }
+            }
+
+            var pollResult = poll.PossibleAnswers
+                .Select(r => (Emoji: r.Emoji, Count: r.Count))
                 .ToList();
-            //var pollResult = await interactivity.DoPollAsync(pollMessage, emojiMapping.Keys.ToArray(), PollBehaviour.KeepEmojis, pollDuration);
+
             await pollMessage.UnpinAsync();
             if (!pollResult.Any(r => r.Count > 0))
             {
                 await channel.SendMessageAsync($"No one participated in the poll {poll.Question} :(");
                 return;
             }
+
+            Dictionary<DiscordEmoji, string> emojiMapping = GetEmojiMapping(poll.PossibleAnswers.Select(x=>x.Value).ToList());
             int totalVotes = pollResult.Sum(r => r.Count);
             var pollResultEmbed = new DiscordEmbedBuilder()
                 .WithTitle($"Poll results: {poll.Question}")
                 .WithColor(DiscordColor.Azure)
-                .WithDescription($"**{pollCreator.Mention}{(pollCreator.Username.EndsWith('s') ? "'" : "'s")} poll ended. Here are the results:**\n\n" +
-                string.Join("\n", emojiMapping
-                    .Select(ans => new { Answer = ans.Value, Votes = pollResult.Single(r => r.Emoji == ans.Key).Count })
-                    .Select(ans => $"**{ans.Answer}**: {"vote".ToQuantity(ans.Votes)} ({100.0 * ans.Votes / totalVotes:F1} %)"))
+                .WithDescription(
+                    $"**{pollCreator.Mention}{(pollCreator.Username.EndsWith('s') ? "'" : "'s")} poll ended. Here are the results:**\n\n" +
+                    string.Join("\n", emojiMapping
+                        .Select(ans =>
+                            new {Answer = ans.Value, Votes = pollResult.Single(r => r.Emoji == ans.Key).Count})
+                        .Select(ans =>
+                            $"**{ans.Answer}**: {"vote".ToQuantity(ans.Votes)} ({100.0 * ans.Votes / totalVotes:F1} %)"))
                 );
             await channel.SendMessageAsync(embed: pollResultEmbed.Build());
```

Comment:
At this point we can also remove the "vote buttons" from the poll. Maybe also update the footer with poll ended